packages/framework/immutable-merge/src/Merge.ts

```diff
@@ -51,76 +128,78 @@ export type IMergeRecursionHandler = (options: IMergeOptions, ...objs: any[]) =>
  * be `{}` - no recursion, `{ depth: -1 }` - recurse infinitely
  * @param objs - an array of objects to merge together
  */
-export function immutableMergeCore(options: IMergeOptions, ...objs: any[]): any {
-  const setToMerge = objs.filter(v => _isNonArrayObject(v, true));
-  const { depth = 0, recurse } = options;
-  const processSingle = options.processSingles && setToMerge.length === 1;
+function immutableMergeWorker(mergeOptions: RecursionOption | MergeOptions, singleMode: boolean, ...objs: any[]): any {
```

Comment:
I remember there being a discussion on why it was very difficult to type parameterize some of these functions, requiring the use of `any`, but I don't quite remember the details. Can you expand on that a bit?

Reply:
I don't remember many of the issues, the public facing functions are not typed to any as they expect sets of object and have a type to coerce similarity between inputs (if someone is using inference)

Realistically though, the internal worker function use of any is somewhat accurate as we are walking and type checking against various object types, including filtering sets of inputs to only merge ones that are object based.

Reply:
I just checked again on the any vs object core type.  I changed the internal type to a generic and it looks like the inferencing chains correctly.